src/holofuel/pages/TransactionHistory/TransactionHistory.js

```diff
@@ -94,16 +107,24 @@ const TransactionTableHeading = ({ content }) => {
   </th>
 }
 
-export function TransactionRow ({ transaction, showCancellationModal, completed }) {
+export function TransactionRow ({ transaction, showCancellationModal, completed, counterpartyList }) {
   const { id, timestamp, amount, counterparty, direction, fees, presentBalance, notes } = transaction
+
+  let counterpartyNick = 'Loading...'
+  if (counterpartyList) counterpartyNick = counterpartyList.find(agent => agent.id === counterparty).nickname
```

Comment:
I think there's a way we can avoid this using Apollo Cache. We can have `useFetchCounterparties` update the cache so that the nickname is in `Transaction.user`. I just looked in `Schema.graphql` and currently `Transaction.counterparty` is of type String, and there's no field of type `HolofuelUser` on `Transaction`. I think part of this will be adding that field.

Reply:
Interesting idea. I'l look into this. :+1:

Reply:
Done.